pkg/frontend/querymiddleware/astmapper/instant_splitting.go

```diff
@@ -64,12 +64,17 @@ var splittableRangeVectorAggregators = map[string]bool{
 
 // NewInstantQuerySplitter creates a new query range mapper.
 func NewInstantQuerySplitter(interval time.Duration, logger log.Logger) ASTMapper {
-	return NewASTExprMapper(
+	instantQueryMapper := NewASTExprMapper(
 		&instantSplitter{
 			interval: interval,
 			logger:   logger,
 		},
 	)
+
+	return NewMultiMapper(
+		instantQueryMapper,
+		newSubtreeFolder(),
```

Comment:
Not really following the logic of adding `subrtreeFolder` mapper based on the explanation:
> The query-frontend is not capable of fetching any samples to run a PromQL query directly from ingesters or store-gateways. This means that every single vector selector needs to be pushed down to queriers. The way this is implemented in the query sharding is using subtreeFolder. This PR fixes it using it for instant queries splitting too.

What does the subtree mapper do and why is this related to querying ingesters and store-gateways?

Reply:
It look for any vector selector and, if it's not already part of query which will be pushed down to queriers, it will wrap it with `__embedded_queries__` to push it down.